src/main/java/org/springframework/retry/policy/SimpleRetryPolicy.java

```diff
@@ -136,72 +157,12 @@ public void setMaxAttempts(int maxAttempts) {
 	 * @return the maximum number of attempts
 	 */
 	public int getMaxAttempts() {
-		return this.maxAttempts;
-	}
-
-	/**
-	 * Test for retryable operation based on the status.
-	 *
-	 * @see org.springframework.retry.RetryPolicy#canRetry(org.springframework.retry.RetryContext)
-	 *
-	 * @return true if the last exception was retryable and the number of
-	 * attempts so far is less than the limit.
-	 */
-	@Override
-	public boolean canRetry(RetryContext context) {
-		Throwable t = context.getLastThrowable();
-		return (t == null || retryForException(t)) && context.getRetryCount() < maxAttempts;
-	}
-
-	/**
-	 * @see org.springframework.retry.RetryPolicy#close(RetryContext)
-	 */
-	@Override
-	public void close(RetryContext status) {
-	}
-
-	/**
-	 * Update the status with another attempted retry and the latest exception.
-	 *
-	 * @see RetryPolicy#registerThrowable(RetryContext, Throwable)
-	 */
-	@Override
-	public void registerThrowable(RetryContext context, Throwable throwable) {
-		SimpleRetryContext simpleContext = ((SimpleRetryContext) context);
-		simpleContext.registerThrowable(throwable);
-	}
-
-	/**
-	 * Get a status object that can be used to track the current operation
-	 * according to this policy. Has to be aware of the latest exception and the
-	 * number of attempts.
-	 *
-	 * @see org.springframework.retry.RetryPolicy#open(RetryContext)
-	 */
-	@Override
-	public RetryContext open(RetryContext parent) {
-		return new SimpleRetryContext(parent);
-	}
-
-	private static class SimpleRetryContext extends RetryContextSupport {
-		public SimpleRetryContext(RetryContext parent) {
-			super(parent);
-		}
-	}
```

Comment:
Previously, there was `private static class SimpleRetryContext extends RetryContextSupport` that brings nothing new, and I've replaced it by direct usage of RetryContextSupport. Did the presence of SimpleRetryContext have any secret considerations?

Reply:
Don't know. Probably in practice not, but I think we should keep it - you can't assume that no-one ever used it, and we don't like to break existing apps.

Reply:
I'm also thinking how to not break something unknown, that may access this private structures.
The most safe way is to keep SimpleRetryPolicy totally unchanged. (even not to make it CompositeRetryPolicy's subclass).
The flip side is a logic duplication in SimpleRetryPolicy and MaxAttemptsRetryPolicy.
We can buy more safety by this cost.
What do you think?

Reply:
I've added new commit where SimpleRetryPolicy structure got back, except obtaining default classifier instance from BinaryExceptionClassifier.